opba-protocols/xs2a-protocol-tests/xs2a-bdd-wiremock/src/test/java/de/adorsys/opba/protocol/xs2a/tests/e2e/wiremock/WiremockE2EXs2aProtocolTest.java

```diff
@@ -139,4 +140,56 @@ void testTransactionsListWithConsentUsingEmbedded(Approach approach) {
                     WiremockConst.MAX_MUSTERMAN_RESOURCE_ID, WiremockConst.DATE_FROM, WiremockConst.DATE_TO, WiremockConst.BOTH_BOOKING
                 );
     }
+
+    @Test
+    void testAccountsListWithConsentUsingRedirectWithIpAddress() {
+        given()
+                .redirect_mock_of_sandbox_for_anton_brueckner_accounts_running()
+                .preferred_sca_approach_selected_for_all_banks_in_opba(Approach.REDIRECT)
+                .rest_assured_points_to_server();
+
+        when()
+                .fintech_calls_list_accounts_for_anton_brueckner()
+                .and()
+                .user_anton_brueckner_provided_initial_parameters_to_list_accounts_with_all_accounts_consent_with_ip_address_check()
+                .and()
+                .open_banking_redirect_from_aspsp_ok_webhook_called();
+        then()
+                .open_banking_has_consent_for_anton_brueckner_account_list()
+                .open_banking_can_read_anton_brueckner_account_data_using_consent_bound_to_service_session();
+    }
+
+    @Test
+    void testAccountsListWithConsentUsingRedirectWithComputedIpAddress() {
+        given()
+                .redirect_mock_of_sandbox_for_anton_brueckner_accounts_running()
+                .preferred_sca_approach_selected_for_all_banks_in_opba(Approach.REDIRECT)
+                .rest_assured_points_to_server();
+
+        when()
+                .fintech_calls_list_accounts_for_anton_brueckner_ip_address_compute()
+                .and()
+                .user_anton_brueckner_provided_initial_parameters_to_list_accounts_with_all_accounts_consent_with_ip_address_check()
+                .and()
+                .open_banking_redirect_from_aspsp_ok_webhook_called();
+        then()
+                .open_banking_has_consent_for_anton_brueckner_account_list()
+                .open_banking_can_read_anton_brueckner_account_data_using_consent_bound_to_service_session();
+    }
+
+    @Test
+    void testAccountsListWithConsentUsingRedirectWithoutIpAddress() {
+        given()
+                .redirect_mock_of_sandbox_for_anton_brueckner_accounts_running()
+                .preferred_sca_approach_selected_for_all_banks_in_opba(Approach.REDIRECT)
+                .rest_assured_points_to_server();
+
+        when()
+                .fintech_calls_list_accounts_for_anton_brueckner_no_ip_address()
+                .and()
+                .user_anton_brueckner_provided_initial_parameters_to_list_accounts_with_all_accounts_consent_with_exception();
+        then()
+                .an_exception_is_thrown();
```

Comment:
Instead of `an_exception_is_thrown` it is better to have following:
```
given()
		.redirect_mock_of_sandbox_for_anton_brueckner_accounts_running()
		.preferred_sca_approach_selected_for_all_banks_in_opba(Approach.REDIRECT)
		.rest_assured_points_to_server();
then()
		.fintech_calls_list_accounts_for_anton_brueckner_no_ip_address_and_gets_5xx();
```

And
fintech_calls_list_accounts_for_anton_brueckner_no_ip_address_and_gets_5xx();
is
```
		ExtractableResponse<Response> response = withHeadersWithoutIpAddress(ANTON_BRUECKNER)
					.header(SERVICE_SESSION_ID, UUID.randomUUID().toString())
					.header(COMPUTE_PSU_IP_ADDRESS, false)
				.when()
					.get(AIS_ACCOUNTS_ENDPOINT)
                .then()
					.statusCode(oneOf(500, 503, 504))
```

Reply:
but it doesn't return oneOf(500, 503, 504). It returns 20x on AIS_ACCOUNTS_ENDPOINT and 303 on AUTHORIZE_CONSENT_ENDPOINT.
As I understand it will endless try to get missing field - ip address